app/src/main/java/com/github/palFinderTeam/palfinder/meetups/MeetupListAdapter.kt

```diff
@@ -0,0 +1,53 @@
+package com.github.palFinderTeam.palfinder.meetups
+
+import android.annotation.SuppressLint
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Filter
+import android.widget.Filterable
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.github.palFinderTeam.palfinder.R
+import com.github.palFinderTeam.palfinder.utils.PrettyDate
+import com.github.palFinderTeam.palfinder.utils.searchedFilter
+
+
+class MeetupListAdapter<T : MeetUp>(private val dataSet : List<MeetUp>): RecyclerView.Adapter<MeetupListAdapter.ViewHolder>(), Filterable {
+    val currentDataSet = dataSet.toMutableList()
+
+    class ViewHolder(view : View) : RecyclerView.ViewHolder(view) {
+        //TODO - add some remaining fields to display
+        val meetup_title: TextView = view.findViewById(R.id.meetup_title)
+        val meetup_date: TextView = view.findViewById(R.id.date)
+        val meetup_description: TextView = view.findViewById(R.id.meetup_description)
+        val meetup_number_participants: TextView = view.findViewById(R.id.number_participants)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, type: Int): ViewHolder {
+        //create a new view for each meetup
+        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.meetup_listview, parent, false))
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        //update displayed elements with the content of the current dataset
+        val meetup_title = holder.meetup_title
+        meetup_title.text = currentDataSet[position].name
+        val meetup_date = holder.meetup_date
+        val prettyDate = PrettyDate()
+        meetup_date.text = "in" + prettyDate.timeSince(currentDataSet[position].startDate.time)

```

Comment:
Check the new version of PrettyDate, it works now with forward dates. Also no need to use the prefix "in", "in" and "ago" likewise are generated automatically

Reply:
I think the version I have is not the newest, so I will update it as soon as possible